lib/kube/proxy/watcher.go

```diff
@@ -248,3 +248,17 @@ func (s *TLSServer) deleteKubernetesServer(ctx context.Context, name string) err
 	}
 	return nil
 }
+
+// startKubeServerResourceWatcher starts watching changes to Kube servers resources.
+func (s *TLSServer) startKubeServerResourceWatcher(ctx context.Context) (*services.KubeServerWatcher, error) {
```

Comment:
No.
The idea was to simplify the logic since the watcher is only required if the running mode is `ProxyService` or `LegacyProxyService`.

Do you prefer that we receive the watcher from the outside?

Reply:
Changed in ec13e49068a650d64b89ae025acec6843987e749